### ISSUE TITLE
Leftover key transformation?

### ISSUE THREAD

**Issue by Propaganistas** [adamwathan/form#87]
Hi

When trying to debug [this issue](https://github.com/Propaganistas/Laravel-Translatable-Bootforms/issues/6) at TranslatableBootForms, I stumbled upon the following code which seems to be the underlying cause:

`FormBuilder.php` line 222:
```php
public function getValueFor($name)
{
    $name = $this->transformKey($name);

    if ($this->hasOldInput()) {
        return $this->getOldInput($name);
    }

    // ...
}
//...
protected function getOldInput($name)
{
    return $this->escape($this->oldInput->getOldInput($name));
}
//...
protected function transformKey($key)
{
    return str_replace(['.', '[]', '[', ']'], ['_', '', '.', ''], $key);
}
```

`IlluminateOldInputProvider.php` line 21:
```php
public function getOldInput($key)
{
    return $this->session->getOldInput($this->transformKey($key));
}
//...
protected function transformKey($key)
{
    return str_replace(['.', '[]', '[', ']'], ['_', '', '.', ''], $key);
}
```
Is there any specific reason why `transformKey()` is called twice throughout this process (in `getValueFor()` and `getOldInput()`)?
Right now, the following transformations will occur to e.g. `en[title]`:
`en[title] --> en.title --> en_title`
As you can see, there seems to be a superfluous transformation since `en.title` should be the key to search for in old input session store.

**Comment by adamwathan** [comment]
@JesseLeite Any thoughts on this? Looks like that  was introduced in this PR: https://github.com/adamwathan/form/pull/78

**Comment by JesseLeite** [comment]
Oh no ...git blamed! I'll look into it.

**Comment by adamwathan** [comment]
Fixed and tagged v0.8.3! Thanks.

**Comment by Propaganistas** [comment]
Thanks for the quick follow-up!
